src/runner/pythonrunner.cpp

```diff
@@ -994,30 +981,61 @@ BOOST_PYTHON_MODULE(mobase)
-      .def("setGameName", bpy::pure_virtual(&IModInterface::setGameName))
+      .def("setGamePlugin", bpy::pure_virtual(&IModInterface::setGamePlugin))
       .def("setName", bpy::pure_virtual(&IModInterface::setName))
       .def("remove", bpy::pure_virtual(&IModInterface::remove))
       ;
 
   bpy::enum_<MOBase::EGuessQuality>("GuessQuality")
-      .value("invalid", MOBase::GUESS_INVALID)
-      .value("fallback", MOBase::GUESS_FALLBACK)
-      .value("good", MOBase::GUESS_GOOD)
-      .value("meta", MOBase::GUESS_META)
-      .value("preset", MOBase::GUESS_PRESET)
-      .value("user", MOBase::GUESS_USER)
+      .value("INVALID", MOBase::GUESS_INVALID)
+      .value("FALLBACK", MOBase::GUESS_FALLBACK)
+      .value("GOOD", MOBase::GUESS_GOOD)
+      .value("META", MOBase::GUESS_META)
+      .value("PRESET", MOBase::GUESS_PRESET)
+      .value("USER", MOBase::GUESS_USER)
       ;
 
   bpy::class_<MOBase::GuessedValue<QString>, boost::noncopyable>("GuessedString")
+      .def(bpy::init<>())
+      .def(bpy::init<QString const&, EGuessQuality>())
+      .def("update",
+           static_cast<GuessedValue<QString>& (GuessedValue<QString>::*)(const QString&)>(&GuessedValue<QString>::update),
+           bpy::return_self<>())
       .def("update",
-           static_cast<GuessedValue<QString> &(GuessedValue<QString>::*)(const QString&, EGuessQuality)>(&GuessedValue<QString>::update),
-           bpy::return_value_policy<bpy::reference_existing_object>(), updateWithQuality())
-      .def("variants", &MOBase::GuessedValue<QString>::variants, bpy::return_value_policy<bpy::copy_const_reference>())
+           static_cast<GuessedValue<QString>& (GuessedValue<QString>::*)(const QString&, EGuessQuality)>(&GuessedValue<QString>::update),
+           bpy::return_self<>())
+
+      // Methods to simulate the assignment operator:
+      .def("reset", +[](GuessedValue<QString>* gv) { *gv = GuessedValue<QString>(); }, bpy::return_self<>())
+      .def("reset", +[](GuessedValue<QString>* gv, const QString& value, EGuessQuality eq) { *gv = GuessedValue<QString>(value, eq); }, bpy::return_self<>())
+      .def("reset", +[](GuessedValue<QString>* gv, const GuessedValue<QString>& other) { *gv = other; }, bpy::return_self<>())
+
+      // Use an intermediate lambda to avoid having to register the std::function conversion:
+      .def("setFilter", +[](GuessedValue<QString>* gv, std::function<std::variant<QString, bool>(QString const&)> fn) {
+        gv->setFilter([fn](QString& s) {
+          auto ret = fn(s);
+          return std::visit([&s](auto v) {
+            if constexpr (std::is_same_v<decltype(v), QString>) {
+              s = v;
+              return true;
+            }
+            else if constexpr (std::is_same_v<decltype(v), bool>) {
+              return v;
+            }
+            else {
+              static_assert("Incorrect visitor.");
+            }
+          }, ret);
+        });
+      })
```

Comment:
This looks complicated. In other places where something's taken a `std::function`, the functor converter has been fine.

Reply:
That's different. We may discuss this further on the Discord because this will appear in various places.

The issue here is that the functor takes a reference. If we use the standard converter, it is not possible to provide such callable from python. The way around this is to have the python callback return a string in case the string has been modified. 

Examples:
```cpp
name.setFilter([](QString &name) {
    if (name.endswith(".pdf")) {
        return false;
    }
    if (name.endswith(".txt")) {
        name = txtFile; // Modify the reference.
    }
    return true;
});
```

In python:
```python
def filter_fn(name):
    if name.endswith(".pdf"):
        return False

    if name.endswith(".txt"):
        return "TXT_FILE"  # Returns a new string (cannot modify name)

    return True

name.setFilter(filter_fn)
```

Reply:
That makes sense. I think I managed to get something working somewhere else where Python modified a reference, but it wasn't with a function and I remember it being more of a pain than it had any legitimate reason to be.

Reply:
You had stuff for `GuessedString`, I don't know if this was working or not, but from a python point-of-view it was very... weird (assigning a reference that modify the original object). I've replaced it with the `.reset()` method which feel more pythonic.

In multiple places where there were references or pointer as argument in C++, I went for this kind of stuff in python (i.e., returning a variant).

The `register_variant` and `register_tuple` are here to support that. It may be a bit more writing on the C++ side sometimes, but it is (I think) quite simple code, and makes the python side much nicer.

Reply:
The focus was on having the same signatures as the C++ version because the UIBase headers were the only API documentation available, and where necessary, being Pythonic was completely disregarded. Your `.pyi` work will change the priorities there a lot.